main_test.go

```diff
@@ -1,8 +1,88 @@
 package main
 
 import (
+	"compress/gzip"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"strconv"
+	"strings"
 	"testing"
+
+	corev2 "github.com/sensu/sensu-go/api/core/v2"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
 )
 
+func splitTestURL(t *testing.T, url string) (string, int) {
+	split := strings.Split(url, ":")
+	require.Equal(t, 3, len(split))
+	host := fmt.Sprintf("%s:%s", split[0], split[1])
+	i, err := strconv.Atoi(split[2])
+	require.NoError(t, err)
+	port := i
+	return host, port
+}
```

Comment:
For next time ;)
https://golang.org/pkg/net/url/#URL
https://golang.org/pkg/net/url/#ParseRequestURI

Reply:
It appears a URL still has the host and port in the same string?
```
Host       string    // host or host:port
```

Reply:
Sorry I wasn't very clear; I meant you can build a `url.URL` with `ParseRequestURI()`, then the following member functions give you the host name and port respectively:
- https://golang.org/pkg/net/url/#URL.Hostname
- https://golang.org/pkg/net/url/#URL.Port